storage/leveldb/common.go

```diff
@@ -0,0 +1,32 @@
+package leveldb
+
+import (
+	"fmt"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+)
+
+func openLevelDB(path string, options *opt.Options) (*leveldb.DB, error) {
+	db, errOpen := leveldb.OpenFile(path, options)
+	if errOpen != nil {
```

Comment:
this may be a different error, such as file already opened
in this case recover should not be done

Reply:
https://godoc.org/github.com/syndtr/goleveldb/leveldb#Open

Open will return an error with type of ErrCorrupted if corruption detected in the DB. Use errors.IsCorrupted to test whether an error is due to corruption. Corrupted DB can be recovered with Recover function

Reply:
done, added check